actors/builtin/miner/miner_actor.go

```diff
@@ -889,22 +832,38 @@ func (a Actor) verifySurprisePost(rt Runtime, st *State, onChainInfo *abi.OnChai
 	}
 
 	// verify appropriate number of tickets is present
-	if int64(len(onChainInfo.Candidates)) != NumSurprisePoStSectors {
+	if int64(len(onChainInfo.Candidates)) != NumFallbackPoStSectors {
 		rt.Abort(exitcode.ErrIllegalArgument, "Invalid Surprise PoSt. Too few tickets included.")
 	}
 
-	randomnessK := rt.GetRandomness(challengeEpoch - PoStLookback)
+	randomnessK := rt.GetRandomness(challengeEpoch)
+
 	// regenerate randomness used. The PoSt Verification below will fail if
 	// the same was not used to generate the proof
-	postRandomness := crypto.DeriveRandWithMinerAddr(crypto.DomainSeparationTag_SurprisePoStChallengeSeed, randomnessK, rt.CurrReceiver())
+	postRandomness := crypto.DeriveRandWithMinerAddr(crypto.DomainSeparationTag_FallbackPoStChallengeSeed, randomnessK, rt.CurrReceiver())
+
+	store := adt.AsStore(rt)
+	provingSet := adt.AsArray(store, st.ProvingSet)
+
+	var sectorInfos []abi.SectorInfo
+	var ssinfo SectorOnChainInfo
+	provingSet.ForEach(&ssinfo, func(i int64) error {
+
+		// TODO: faults!!!!
```

Comment:
if any one of the sector here fails, miner enters Miner-Wide fault, power goes to zero.

Reply:
@zixuanzh that doesn't make sense. 

I think @whyrusleeping 's comment is noting that declared faulty sectors need to be excluded from the proving set.

However, if this proof verification fails, _nothing should happen_. The miner has until the end of the window to submit a good proof.

Reply:
Some in https://github.com/filecoin-project/specs-actors/pull/143